RecoLocalCalo/HGCalRecProducers/plugins/HGCalLayerClusterProducer.cc

```diff
@@ -194,40 +197,45 @@ void HGCalLayerClusterProducer::produce(edm::Event& evt, const edm::EventSetup&
 
   edm::PtrVector<reco::BasicCluster> clusterPtrs, clusterPtrsSharing;
 
-  std::vector<float> times;
+  std::vector<std::pair<float, float>> times;
   times.reserve(clusterHandle->size());
 
   for (unsigned i = 0; i < clusterHandle->size(); ++i) {
     edm::Ptr<reco::BasicCluster> ptr(clusterHandle, i);
     clusterPtrs.push_back(ptr);
 
-    float timeCl = -99.;
+    std::pair<float, float> timeCl(-99., -1.);
+
     const reco::CaloCluster& sCl = (*clusterHandle)[i];
-    if (sCl.size() >= 3) {
+    if (sCl.size() >= nHitsTime) {
       std::vector<float> timeClhits;
+      std::vector<float> timeErrorClhits;
 
       for (auto const& hit : sCl.hitsAndFractions()) {
         auto finder = hitmap.find(hit.first);
         if (finder == hitmap.end())
           continue;
 
         //time is computed wrt  0-25ns + offset and set to -1 if no time
-        float rhTime = finder->second;
-        if (rhTime < 0.)
+        const HGCRecHit* rechit = finder->second;
+        float rhTimeE = rechit->timeError();
+        //check on timeError to exclude scintillator
+        if (rhTimeE < 0.)
           continue;
-        timeClhits.push_back(rhTime - timeOffset);
+        timeClhits.push_back(rechit->time() - timeOffset);
+        timeErrorClhits.push_back(1. / (rhTimeE * rhTimeE));
       }
-      if (timeClhits.size() >= 3)
-        timeCl = hgcalsimclustertime::fixSizeHighestDensity(timeClhits);
+      hgcalsimclustertime::ComputeClusterTime timeEstimator;
+      timeCl = timeEstimator.fixSizeHighestDensity(timeClhits, timeErrorClhits, nHitsTime);
```

Comment:
This is the only case in which the ```minNhits``` parameter of ```fixSizeHighestDensity``` is configurable: in all other calls of ```fixSizeHighestDensity``` the default value of "3" is used instead. 
This means that whenever "nHitsTime" is set to something different from "3" in the ```hgcalLayerClusters``` config, then the different calls of ```fixSizeHighestDensity``` would use a different value for ```minNhits```.
Please confirm that this is the wanted behaviour

Reply:
Yes, confirmed